src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1573,38 +1723,48 @@ private void ImportFallthrough(BasicBlock next)
                 if (_stack.Length > 0)
                 {
                     entryStack = new EvaluationStack<StackEntry>(_stack.Length);
-
-#pragma warning disable 162 // Due to not implement3ed exception incrementer in for needs pragma warning disable
                     for (int i = 0; i < _stack.Length; i++)
                     {
-                        // todo: do we need anything special for spilled stacks like cpp codegen does?
-                        entryStack.Push(_stack[i]);
-                        //entryStack.Push(NewSpillSlot(_stack[i]));
+                        entryStack.Push(NewSpillSlot(_stack[i]));
                     }
-#pragma warning restore 162
                 }
                 next.EntryStack = entryStack;
             }
 
             if (entryStack != null)
             {
-                // todo: do we have to do anything here?
-#pragma warning disable 162// Due to not implement3ed exception incrementer in for needs pragma warning disable
                 for (int i = 0; i < entryStack.Length; i++)
                 {
-                    /*AppendLine();
-                    Append(entryStack[i]);
-                    Append(" = ");
-                    Append(_stack[i]);
-                    AppendSemicolon();*/
+                    var currentEntry = _stack[i];
+                    var entry = entryStack[i] as SpilledExpressionEntry;
+                    if (entry == null)
+                        throw new InvalidProgramException();
+
+                    if (currentEntry is SpilledExpressionEntry)
+                        continue; //this is already a sharable value
+
+                    StoreTemp(entry.LocalIndex, currentEntry.ValueAsType(entry.Type, this));
```

Comment:
Should we consider using LLVM phis instead of storing temps?

Reply:
This seems like it would be a much better idea, but I'm not 100% sure how to fit it into the basic blocks stuff. The main issue blocking the straightforward implementation is I'm not sure the block generation is ordered in such a way that we always know where we're coming from at the start of the basic block. If that's the case we'll have to either rewrite the header of affected basic blocks, or do a separate pass to determine the spilled stack entries before importing the instructions.

Reply:
That's fine. I was just curious if there was a reasonable way to handle it. Spilling might actually be something we need for GC tracking and other cases.